assets/templates/app/drivers/device.js

```diff
@@ -3,11 +3,47 @@
 const Homey = require('homey');
 
 class MyDevice extends Homey.Device {
-	
-	onInit() {
-		this.log('MyDevice has been inited');
-	}
-	
+  /**
+   * onInit is called when the device is initialized.
+   */
+  async onInit() {
+    this.log('MyDevice has been initialized');
+  }
+
+  /**
+   * onAdded is called when the user adds the device, called just after pairing.
+   */
+  onAdded() {
+    this.log('MyDevice has been added');
```

Comment:
Should we add `super.onAdded()` here? If this app uses a library (e.g. homey-zigbeedriver) which adds functionality by overriding `onAdded` itself stuff breaks.

Reply:
Technically homey-zigbeedriver could override `_onAdded` so that user-code just works™️.
On the other hand we could teach devs to always call super and do away with the underscore prefixed methods...

Reply:
zwavedriver and oauth2app use `onMeshAdded` respectively `onOAuth2Added` for this.

Reply:
Yet this would do no harm:

```
async onAdded() {
   await super.onAdded();

   this.log('MyDevice has been added');
}
```

Reply:
It feels arbitrary/confusing to only do this for `Device#onAdded` but it also feels weird to do this for all overloaded methods when in most cases it isn't necessary...

```js
  /**
   * onInit is called when the device is initialized.
   */
  async onInit() {
    await super.onInit();

    this.log('MyDevice has been initialized');
  }


  /**
   * onAdded is called when the user adds the device, called just after pairing.
   */
  async onAdded() {
    await super.onAdded();

    this.log('MyDevice has been added');
  }

  /**
   * onSettings is called when the user updates the device's settings.
   * @param {object} data the onSettings event data
   * @param {object} data.oldSettings The old settings object
   * @param {object} data.newSettings The new settings object
   * @param {string[]} data.changedKeys An array of keys changed since the previous version
   * @returns {Promise<string|void>} return a custom message that will be displayed
   */
  async onSettings(data) {
    await super.onSettings(data);

    this.log('MyDevice settings where changed');
  }

  /**
   * onRenamed is called when the user updates the device's name.
   * This method can be used this to synchronise the name to the device.
   * @param {string} name The new name
   */
  async onRenamed(name) {
    await super.onRenamed(name);

    this.log('MyDevice was renamed');
  }

  /**
   * onDeleted is called when the user deleted the device.
   */
  async onDeleted() {
    await super.onDeleted();

    this.log('MyDevice has been deleted');
  }
```

Reply:
You are right, not sure what is best. Maybe just leave super out, by default no lib is used so the default does not need super I guess.

Reply:
since we know what drivers/plugins someone will install we can also include multiple templates where we can add the super call 🤔 . Out of scope for this PR though

Reply:
Agreed, will merge and release!